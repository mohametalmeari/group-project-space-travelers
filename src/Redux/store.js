import { configureStore } from '@reduxjs/toolkit';
import missionReducer from './features/missions/missionSlice';

const store = configureStore({
  reducer: {
    mission: missionReducer,
  },
});

export default store;
