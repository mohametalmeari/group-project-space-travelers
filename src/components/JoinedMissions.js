import { useSelector } from 'react-redux';

const JoinedMissions = () => {
  const { missions } = useSelector((state) => state.mission);
  return (
    <>
      <h2>My Missions</h2>
      <ul>
        {missions.map((item) => (
          item.reserved
            && (
            <li key={item.id}>
              {item.name}
            </li>
            )
        ))}
      </ul>
    </>
  );
};
export default JoinedMissions;
