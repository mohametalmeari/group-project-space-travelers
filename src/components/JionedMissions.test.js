import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../Redux/store';
import JoinedMissions from './JoinedMissions';

describe('JoinedMissions Component', () => {
  test('Test render of JoinedMissions', () => {
    const { container } = render(
      <MemoryRouter>
        <Provider store={store}>
          <JoinedMissions />
        </Provider>
      </MemoryRouter>,
    );
    expect(container).toMatchSnapshot();
  });
});
