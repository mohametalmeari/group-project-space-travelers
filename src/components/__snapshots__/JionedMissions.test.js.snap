// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JoinedMissions Component Test render of JoinedMissions 1`] = `
<div>
  <h2>
    My Missions
  </h2>
  <ul />
</div>
`;

exports[`Layout Component Test render of Layout 1`] = `
<div>
  <h2>
    My Missions
  </h2>
  <ul />
</div>
`;
